src/store/modules/game.js

```diff
@@ -350,7 +350,17 @@ export default {
         );
       });
     },
-
+    async requestSpectateLeave(context) {
+      return new Promise((resolve, reject) => {
+        io.socket.get('/game/spectateLeave', function handleResponse(res, jwres) {
+          if (jwres.statusCode === 200) {
+            context.commit('resetState');
+            return resolve();
+          }
+          return reject(new Error('Error leaving spectator'));
```

Comment:
What should we do when we get this error? I think the menu should at least navigate us home still so we aren't trapped in a game we don't want to watch anymore. It's okay to throw this error here as long as it's handled above.
```suggestion
          return reject(new Error('Error leaving game as spectator'));
```